katdal/chunkstore_s3.py

```diff
@@ -512,33 +511,38 @@ def request(self, method, url, chunk_name='', ignored_errors=(), timeout=(), **k
         StoreUnavailable
             If a general HTTP error occurred that is not ignored
         """
-        kwargs['timeout'] = self.timeout if timeout is () else timeout
+        kwargs['timeout'] = self.timeout if timeout == () else timeout
+
+        def _error_msg(chunk_name, response):
+            prefix = 'Chunk {!r}: '.format(chunk_name) if chunk_name else ''
+            return '{}Store responded with HTTP error {} ({}) to request: {} {}'.format(
+                prefix, response.status_code, response.reason,
+                response.request.method, response.url)
+
         # Use _standard_errors to filter errors emanating from within with-block
         with self._standard_errors(chunk_name), self._session_pool() as session:
             with session.request(method, url, **kwargs) as response:
                 # Turn error responses into the appropriate exception, like raise_for_status
                 status = response.status_code
-                prefix = 'Chunk {!r}: '.format(chunk_name) if chunk_name else ''
-                msg = '{}Store responded with HTTP error {} ({}) to request: {} {}'.format(
-                    prefix, status, response.reason, response.request.method, response.url)
-                if status == 401:
-                    raise AuthorisationFailed(msg)
-                elif status in (403, 404):
-                    # Ceph RGW returns 403 for missing keys due to our bucket policy
-                    # (see https://tracker.ceph.com/issues/38638 for discussion)
-                    raise S3ObjectNotFound(msg)
-                elif self._retries.is_retry(method, status):
-                    raise S3ServerGlitch(msg, status)
-                elif 400 <= status < 600 and status not in ignored_errors:
-                    raise StoreUnavailable(msg)
-                else:
-                    try:
-                        yield response
-                    except TruncatedRead as trunc_error:
-                        # A truncated read is considered a glitch with custom status
-                        glitch_error = S3ServerGlitch(prefix + str(trunc_error),
-                                                      _TRUNCATED_HTTP_STATUS_CODE)
-                        raise_from(glitch_error, trunc_error)
+                if status not in ignored_errors:
```

Comment:
I think if you changed this to `if 400 <= status < 600 and status not in ignored_errors` then you could just create the error message inside the `if` statement rather than needing to have a local function for it (since you'd then always be constructing the message inside the `if` statement.

Reply:
D'oh, yes!